examples/common.c

```diff
@@ -184,6 +184,25 @@ static int match_int_internal(
 	return 1;
 }
 
+int match_bool_arg(int *out, struct args_info *args, const char *opt)
+{
+	const char *found = args->argv[args->pos];
+	int res = 0;
+
+	if (strcmp(found, opt) == 0) {
+		*out = 1;
+		return 1;
+	}
+
+	if (!strncmp(found, "--no-", strlen("--no-")) &&
+	    !strcmp(found + strlen("--no-"), opt + 2)) {
+		*out = 0;
+		res = 1;
+	}
+
+	return res;
```

Comment:
You could avoid the `res` variable by just returning `1` in the previous branch and `0` at the end of this function. That would make the asymmetry between the first and second conditional go away. Not urgent though, you should only address this if you're rerolling anyway